m

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+cd "$(dirname "$0")"
+
+# Ensure submodules are cloned and up to date.
+git submodule update --init
+
+container_name=code-server-dev
+
+# Docker build is very verbose even when everything is cached.
+echo "--- Building $container_name"
+docker build -t $container_name -f ./ci/dev.Dockerfile . > /dev/null
+
+set +e
+container_id=$(docker container inspect --format="{{.Id}}" $container_name 2>/dev/null)
+
+if [ $? -eq "0" ]; then
+    echo "--- Killing $container_name"
```

Comment:
What's the purpose of this?

Reply:
To kill an existing instance of the dev container. much better than having to `exec bash` back into the container and kill all the watch processes

Reply:
Shouldn't the watch process be running in the foreground? Then u can just `ctrl+c`.

Reply:
Not if you get detached from the container, which happens to me all the time. When my ssh loses connection for example.

Reply:
If you get detached, the container should exit. You'd want to pass in `-it` for that to work.